src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -1,198 +1,262 @@
+		const isBrotherhoodInWindow = windows[windowIndex].data.filter(value => value.action.id === this.brotherhood.id).length > 0
+		if (isBrotherhoodInWindow) {
+			return 2
+		}
 
-	override initialise(): void {
-		this.blitzActions = fillActions(BLITZ_SKILLS, this.data)
-		const playerFilter = filter<Event>().source(this.parser.actor.id)
-		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.RIDDLE_OF_FIRE.id), this.onGain)
-		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.RIDDLE_OF_FIRE.id), this.onDrop)
+		if (this.timeBetweenLastWindowInSeconds(windowIndex, windows) > DOUBLE_PERFECT_BALANCE_IN_SECONDS) {
+			return 2
+		}
 
-		this.addEventHook('complete', this.onComplete)
+		const blitzSkillsInLastWindow = this.countBlitzSkills(windows[windowIndex - 1])
+		if (blitzSkillsInLastWindow === 2) {
+			return 1
+		}
+		return 2
 	}
 
-	onCast(event: Events['action']): void {
-		const action = this.data.getAction(event.action)
+	private timeBetweenLastWindowInSeconds(windowIndex: number, windows: Array<HistoryEntry<EvaluatedAction[]>>): number {
+		if (windowIndex === 0) {
+			return 0
+		}
+		return (windows[windowIndex].start - windows[windowIndex].start) / 1000
+	}
+}
 
-		if (action == null) { return }
+class BrotherhoodEvaluator implements WindowEvaluator {
+	private brotherhood: ActionRoot['BROTHERHOOD']
+	private brotherhoodStartHistory: number[]
 
-		// MNK mentors want the oGCDs :angryeyes:
-		if (this.riddle != null) {
-			this.riddle.casts.push(event)
-		}
+	constructor(brotherhood: ActionRoot['BROTHERHOOD'], brotherhoodStartHistory: number[]) {
+		this.brotherhood = brotherhood
+		this.brotherhoodStartHistory = brotherhoodStartHistory
 	}
 
-	private onGain(event: Events['statusApply']): void {
-		if (this.riddle != null) { return }
+	suggest(windows: Array<HistoryEntry<EvaluatedAction[]>>) {
+		const possibleNumberOfBrotherhoods = Math.ceil(windows.length / 2)
+		const brotherhoodUsagesInsideRof = windows.map(previousValue => previousValue.data.filter(value => value.action === this.brotherhood))
+			.filter(value => value.length > 0).length
+		const missedBhInsideRof = possibleNumberOfBrotherhoods - brotherhoodUsagesInsideRof
 
-		this.riddle = new Riddle(event.timestamp, this.data)
-		this.riddle.rushing = this.data.statuses.RIDDLE_OF_FIRE.duration >= (this.parser.pull.timestamp + this.parser.pull.duration) - event.timestamp
+		return new TieredSuggestion({
+			icon: this.brotherhood.icon,
+			tiers: SEVERITIES.BROTHERHOOD_OUTSIDE_ROF_SEVERITY,
+			value: missedBhInsideRof,
+			content: <Trans id="mnk.rof.suggestions.brotherhood.content">Missed {missedBhInsideRof} <ActionLink
+				action="BROTHERHOOD"/> inside <ActionLink action="RIDDLE_OF_FIRE"/> window.
+			</Trans>,
+			why: <Trans id="mnk.rof.suggestions.brotherhood.why"><ActionLink
+				action="BROTHERHOOD"/> has a shorter duration than <ActionLink action="RIDDLE_OF_FIRE"/>.</Trans>,
```

Comment:
This message doesn't really explain why they're being dinged. Also, this is a bit iffy to read, avoid line-breaking mid tag, you can start the line with ActionLink outright, and pop the closing Trans tag on the next line aligned with the `why` field.